app/MainHie.hs

```diff
@@ -98,39 +100,33 @@ main = do
 
 run :: GlobalOpts -> IO ()
 run opts = do
+  hSetBuffering stderr LineBuffering
   let mLogFileName = optLogFile opts
 
       logLevel = if optDebugOn opts
                    then L.DEBUG
                    else L.INFO
 
-  Core.setupLogger mLogFileName ["hie"] logLevel
-
-  projGhcVersion <- getProjectGhcVersion
-  when (projGhcVersion /= hieGhcVersion) $
-    warningm $ "Mismatching GHC versions: Project is " ++ projGhcVersion
-            ++ ", HIE is " ++ hieGhcVersion
+  Core.setupLogger mLogFileName ["hie", "hie-bios"] logLevel
 
   origDir <- getCurrentDirectory
 
   maybe (pure ()) setCurrentDirectory $ projectRoot opts
 
   progName <- getProgName
   logm $  "Run entered for HIE(" ++ progName ++ ") " ++ version
-  d <- getCurrentDirectory
-  logm $ "Current directory:" ++ d
+  logm $ "Current directory:" ++ origDir
   args <- getArgs
   logm $ "args:" ++ show args
 
-  let vomitOptions = defaultOptions { boLogging = BlVomit}
-  let defaultOpts = if optGhcModVomit opts then vomitOptions else defaultOptions
-      -- Running HIE on projects with -Werror breaks most of the features since all warnings
-      -- will be treated with the same severity of type errors. In order to offer a more useful
-      -- experience, we make sure warnings are always reported as warnings by setting -Wwarn
-      biosOptions = defaultOpts { boGhcUserOptions = ["-Wwarn"] }
+  let initOpts = defaultCradleOpts { cradleOptsVerbosity = verbosity }
+      verbosity = if optBiosVerbose opts then Verbose else Silent
+      -- biosLogLevel = if optBiosVerbose opts then L.DEBUG else L.INFO
+
+  -- Core.setupLogger mLogFileName ["hie-bios"] biosLogLevel
 
-  when (optGhcModVomit opts) $
-    logm "Enabling --vomit for ghc-mod. Output will be on stderr"
+  when (optBiosVerbose opts) $
+    logm "Enabling verbose mode for hie-bios. This option currently doesn't do anything."
```

Comment:
Why does it not do anything? Make an issue describing what needs to be done.

Reply:
I would be in favour of removing these manual command line arguments in favour for LSP's [`InitialiseParams.trace`](https://microsoft.github.io/language-server-protocol/specifications/specification-3-15/#initialize)

Reply:
Out of legacy reasons, we had to flags `--vomit` for ghc-mod and `-d` for lsp stuff. However, now hie-bios is also using the same logging meachnism like haskell-lsp, and we set the loggers to the same level. So, We should not remove it right away, because lsp-clients might set `--vomit` and otherwise, all lsp-clients for haskell need to be updated.

Reply:
Due to the feature freeze, I would like to create an issue for this after merge. would that be ok @bubba?